### PR TITLE
Linkage of collection camps to a particular user in their profile

### PR DESCRIPTION
**Pull request by tarunnjoshi**
Linkage of collection camps to a particular user in their profile.

Here is the screenshot how it looks like with it's title when collection camp is authorized
image

### CODE REVIEW

**Review by pokhiii — CHANGES_REQUESTED** [review]

File: wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php [hunk 1]
```diff
@@ -285,6 +286,74 @@ public static function handleAuthorizationEmails(string $op, string $objectName,
...
+      ->execute();
+    
+    $collectionCampActivityId = $optionValues->first()['value'];
+
+    try {
+        $results = \Civi\Api4\Activity::create(FALSE)
+          ->addValue('subject', $collectionCampTitle)
+          ->addValue('activity_type_id', $collectionCampActivityId)
+          ->addValue('status_id', 2)
+          ->addValue('activity_date_time', date('YmdHis'))
```
[the remark this answers is not in the document]
Comment (tarunnjoshi) [reply wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php:348]: Yes it's correct

File: wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php [hunk 2]
```diff
@@ -285,6 +286,74 @@ public static function handleAuthorizationEmails(string $op, string $objectName,
...
+    $optionValues = \Civi\Api4\OptionValue::get(TRUE)
+      ->addWhere('option_group_id:label', '=', 'Activity Type')
+      ->addWhere('label', '=', 'Organize Collection Camp')
+      ->execute();
+    
+    $collectionCampActivityId = $optionValues->first()['value'];
+
+    try {
+        $results = \Civi\Api4\Activity::create(FALSE)
+          ->addValue('subject', $collectionCampTitle)
```
[the remark this answers is not in the document]
Comment (tarunnjoshi) [reply wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php:345]: For this i have followed the below steps:
1. Create a Custom Field Group for Activities (Collection Camp Data)
2. Add a Custom Field for Collection Camp Data
3. Custom field with a name like "Collection Camp ID" and set its data type to "Integer"
4. Save the id through API `->addValue('Collection_Camp_Data.Collection_Camp_ID', $collectionCampId)`

File: wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php [hunk 3]
```diff
@@ -284,6 +285,76 @@ public static function handleAuthorizationEmails(string $op, string $objectName,
...
+      ->addWhere('option_group_id:name', '=', 'activity_type')
+      ->addWhere('label', '=', 'Organize Collection Camp')
+      ->execute();
+
+    $collectionCampActivityId = $optionValues->first()['value'];
+
+    try {
+      $results = Activity::create(FALSE)
+        ->addValue('subject', $collectionCampTitle)
+        ->addValue('activity_type_id', $collectionCampActivityId)
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php:344]:
```suggestion
        ->addValue('activity_type_id:name', 'Organize Collection Camp')
```

File: wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php [hunk 4]
```diff
@@ -284,6 +285,76 @@ public static function handleAuthorizationEmails(string $op, string $objectName,
...
+  /**
+   * Log an activity in CiviCRM.
+   */
+  private static function createCollectionCampOrganizeActivity($contactId, $collectionCampTitle, $collectionCampId) {
+    $optionValues = OptionValue::get(TRUE)
+      ->addWhere('option_group_id:name', '=', 'activity_type')
+      ->addWhere('label', '=', 'Organize Collection Camp')
+      ->execute();
+
+    $collectionCampActivityId = $optionValues->first()['value'];
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php:339]: Let's remove this and directly use the label 'Organize Collection Camp' in the query

File: wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php [hunk 5]
```diff
@@ -284,6 +285,76 @@ public static function handleAuthorizationEmails(string $op, string $objectName,
...
+
+    try {
+      $results = Activity::create(FALSE)
+        ->addValue('subject', $collectionCampTitle)
+        ->addValue('activity_type_id', $collectionCampActivityId)
+        ->addValue('status_id', 2)
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php:345]:
```suggestion
        ->addValue('status_id:name', 'complete') // whatever is the name of the status you want to use
```

File: wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php [hunk 6]
```diff
@@ -284,6 +285,76 @@ public static function handleAuthorizationEmails(string $op, string $objectName,
...
+        ->addValue('activity_date_time', date('YmdHis'))
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php:346]: is this storing the time correctly? shouldn't we use more standard format
```suggestion
        ->addValue('activity_date_time', date('Y-m-d H:i:s'))
```

**Review by pokhiii — APPROVED** [review]

**Pull request merged**